crates/adapters/src/transport/s3/mod.rs

```diff
@@ -23,10 +24,28 @@ pub struct S3InputEndpoint {
 }
 
 impl S3InputEndpoint {
-    pub fn new(config: S3InputConfig) -> Self {
-        Self {
-            config: Arc::new(config),
+    pub fn new(config: S3InputConfig) -> AnyResult<Self> {
+        if !config.no_sign_request {
+            if config.aws_access_key_id.is_none() {
+                bail!("the 'aws_access_key_id' property is missing; either set it or use 'no_sign_request=true' to bypass AWS authentication for a public bucket");
```

Comment:
Would be good to double check if the error the user sees gives enough context for which connector, which table etc is affected. If not, this line needs more info.